src/goto-programs/write_goto_binary.cpp

```diff
@@ -169,15 +171,19 @@ bool write_goto_binary(
 
   if(version < GOTO_BINARY_VERSION)
   {
-    throw invalid_command_line_argument_exceptiont(
-      "version " + std::to_string(version) + " no longer supported",
-      "supported version = " + std::to_string(GOTO_BINARY_VERSION));
+    messaget message{message_handler};
+    message.error() << "version " << version << " no longer supported; "
+                    << "supported version = " << GOTO_BINARY_VERSION
+                    << messaget::eom;
+    return true;
```

Comment:
Using return values rather than exceptions? I'd argue yes for two reasons:
1) It is consistent across reading (which already did this) and writing (which now does this) goto binaries.
2) Using return values to communicate errors makes it easier to build (cross-language) APIs.